AtSheet/explainer.md

```diff
@@ -0,0 +1,225 @@
+standards-track, we will retain this document as an archive and use this section
+to keep the community up-to-date with the most current standards venue and
+content location of future work and discussions.
+
+* This document status: **Active**
+* Expected venue: [CSS Working Group](https://www.w3.org/Style/CSS/)
+* Current version: this document
+
+## Introduction
+When developing web components, web authors often encounter challenges with distributing global styles into shadow roots and sharing styles across different shadow roots. Declarative shadow DOM (DSD) enables creation of shadow DOM without JS, but adding styles to DSD requires the developer to either use JS to put a shared stylesheet into `adoptedStyleSheets`, or to duplicate the styles in a `<style>` element for each component instance.
+
+Additionally, bundling of stylesheets is difficult for developers who are distributing web components. They either need to ship many small stylesheets, or use workarounds like `@import url("data...")` which are suboptimal for performance and don't interact well with other patterns.
+
+We propose an enhancement to allow declaration of new stylesheets via an `@sheet` CSS block, and using existing mechanisims such as `@import` and `<link>` to apply those shared styles to DSDs without the use of Javascript.
+
+We're currently investigating this and [Declarative CSS modules](/ShadowDOM/explainer.md) in parallel, and anticipate that we'll be prioritizing only one of these two in the immediate future.
+
+## Goals
+* Allow the reuse of styles in markup-based shadow DOM without requiring JavaScript.
+* Allow reuse of styles in markup-based shadow DOM without requiring external network requests.
+* Allow web authors to selectively pass in global styles from the parent document. 
+* Allow component authors to bundle their CSS into a single file.
+* Allow named `@sheet` references to fully integrate with existing CSS inclusion methods such as `@import` statements and `<link>` tags.
+
+
+## Non-goals
+Some developers have expressed interest in CSS selectors crossing through the Shadow DOM, as discussed in [issue 909](https://github.com/WICG/webcomponents/issues/909#issuecomment-1977487651). While this scenario is related to sharing styles with Shadow DOM elements, it is solving a different problem and should be addressed separately.
+
+## Proposal - `@sheet`
+Create a new `@sheet` CSS block, for separating style sheets with named identifiers.
+
+All examples use a stylesheet cased as `sheet.css` with the following contents:
+```css
+div {
+  color: blue;
+}
+
+@sheet foo {
+  div {
+    color: red;
+  }
+}
+@sheet bar {
+  div {
+    font-family: sans-serif;
+  }
+}
+```
+
+## Proposal - Importing a specific sheet via `@import`
+```html
+<style>
+  @import sheet("sheet.css#foo");
+</style>
+```
+
+This will import only this rules for `foo` - in this case, the rules for the `:host` selector, and will *not* import any rules from `sheet.css` outside of "foo".
```

Comment:
I don't understand the reference to the `:host` selector here and in the next section.
Isn't it the `  div {
    color: red;
  }` rule that's getting pulled in here?

Reply:
Yep, good catch. We had two sets of examples and I replaced half of them to make them consistent but missed this reference to the example.

Just pushed a fix.